solver/llbsolver/ops/exec.go

```diff
@@ -835,22 +860,30 @@ func (r *cacheRefShare) clone() cache.MutableRef {
 
 func (r *cacheRefShare) release(ctx context.Context) error {
 	if r.main != nil {
-		r.main.mu.Lock()
-		defer r.main.mu.Unlock()
 		delete(r.main.shares, r.key)
 	}
 	return r.MutableRef.Release(ctx)
 }
 
+var cacheRefReleaseHijack func()
+var cacheRefCloneHijack func()
```

Comment:
If we move these to a context value, the tests can be run in parallel.

Reply:
Extra code needed for that isn't probably worth it. `clone()` doesn't even take a context atm.